crates/claims/crates/cose/src/sign1.rs

```diff
@@ -0,0 +1,274 @@
+        Self(bytes)
+    }
+
+    pub fn encode(object: impl Into<CoseSign1>, tagged: bool) -> Self {
+        if tagged {
+            Self(TaggedCborSerializable::to_tagged_vec(object.into()).unwrap())
+        } else {
+            Self(CborSerializable::to_vec(object.into()).unwrap())
+        }
+    }
+
+    pub fn as_compact(&self) -> &CompactCoseSign1 {
+        CompactCoseSign1::new(self.0.as_slice())
+    }
+}
+
+impl Deref for CompactCoseSign1Buf {
+    type Target = CompactCoseSign1;
+
+    fn deref(&self) -> &Self::Target {
+        self.as_compact()
+    }
+}
+
+impl Borrow<CompactCoseSign1> for CompactCoseSign1Buf {
+    fn borrow(&self) -> &CompactCoseSign1 {
+        self.as_compact()
+    }
+}
+
+impl From<CborValue> for CompactCoseSign1Buf {
+    fn from(value: CborValue) -> Self {
+        let mut buffer = Vec::new();
+        ciborium::into_writer(&value, &mut buffer).unwrap();
+        Self(buffer)
+    }
+}
+
+impl AsRef<[u8]> for CompactCoseSign1Buf {
+    fn as_ref(&self) -> &[u8] {
+        self.as_bytes()
+    }
+}
+
+impl From<Vec<u8>> for CompactCoseSign1Buf {
+    fn from(value: Vec<u8>) -> Self {
+        Self(value)
+    }
+}
+
+/// Decoded `COSE_Sign1` object.
+pub struct DecodedCoseSign1<T = ()> {
+    /// Signing bytes.
+    pub signing_bytes: UnsignedCoseSign1<T>,
+
+    /// Signature.
+    pub signature: CoseSignatureBytes,
+}
+
+impl<T> DecodedCoseSign1<T> {
+    pub fn map<U>(self, f: impl FnOnce(T, &[u8]) -> U) -> DecodedCoseSign1<U> {
+        DecodedCoseSign1 {
+            signing_bytes: self.signing_bytes.map(f),
+            signature: self.signature,
+        }
+    }
+
+    pub fn try_map<U, E>(
+        self,
+        f: impl FnOnce(T, &[u8]) -> Result<U, E>,
+    ) -> Result<DecodedCoseSign1<U>, E> {
+        Ok(DecodedCoseSign1 {
+            signing_bytes: self.signing_bytes.try_map(f)?,
+            signature: self.signature,
+        })
+    }
+}
+
+impl From<CoseSign1> for DecodedCoseSign1 {
+    fn from(value: CoseSign1) -> Self {
+        Self {
+            signing_bytes: UnsignedCoseSign1 {
+                protected: value.protected,
+                unprotected: value.unprotected,
+                payload: PayloadBytes::from_bytes(value.payload.unwrap_or_default()),
+            },
+            signature: CoseSignatureBytes(value.signature),
+        }
+    }
+}
+
+impl<T> From<DecodedCoseSign1<T>> for CoseSign1 {
+    fn from(value: DecodedCoseSign1<T>) -> Self {
+        Self {
+            protected: value.signing_bytes.protected,
+            unprotected: value.signing_bytes.unprotected,
+            payload: Some(value.signing_bytes.payload.into_bytes()),
+            signature: value.signature.into_bytes(),
+        }
+    }
+}
+
+/// Payload and bytes.
+///
+/// Stores the payload value as interpreted by the application (type `T`) and
+/// the original payload bytes.
+#[derive(Clone, PartialEq)]
+pub struct PayloadBytes<T = ()> {
+    bytes: Vec<u8>,
```

Comment:
What's the value in keeping the original bytes representation? Is it because we cannot be certain that deserialization -> serialization is transitive?